blockchain/subsidy.go

```diff
@@ -120,18 +123,68 @@ func coinbasePaysTreasury(subsidyCache *standalone.SubsidyCache, tx *dcrutil.Tx,
 	return nil
 }
 
+// checkTreasuryBase checks to see if a given block's treasurybase correctly
+// pays the treasury. This is the new function that uses the treasury base for
+// the payout.
+func checkTreasuryBase(subsidyCache *standalone.SubsidyCache, tx *dcrutil.Tx, height int64, voters uint16, params *chaincfg.Params) error {
+	// Treasury subsidy only applies from block 2 onwards.
+	if height <= 1 {
+		return nil
+	}
+
+	// Treasury subsidy is disabled.
+	if params.BlockTaxProportion == 0 {
+		return nil
+	}
+
+	if len(tx.MsgTx().TxOut) != 2 {
+		// Can't get hit
+		return ruleError(ErrInvalidTreasurybaseTxOutputs,
+			fmt.Sprintf("invalid treasurybase number of outputs: %v",
+				len(tx.MsgTx().TxOut)))
+	}
+
+	treasuryOutput := tx.MsgTx().TxOut[0]
+	if treasuryOutput.Version != 0 {
+		// Can't get hit
+		str := fmt.Sprintf("treasury output version %d is instead of %d",
+			treasuryOutput.Version, 0)
+		return ruleError(ErrInvalidTreasurybaseVersion, str)
+	}
+	if len(treasuryOutput.PkScript) != 1 ||
+		treasuryOutput.PkScript[0] != txscript.OP_TADD {
+		// Can't get hit
+		str := fmt.Sprintf("treasury output script is %x instead of %x",
+			treasuryOutput.PkScript, params.OrganizationPkScript)
+		return ruleError(ErrInvalidTreasurybaseScript, str)
+	}
```

Comment:
It prints the entire script as it should. We want to know if it was or was not `OP_TADD` but also if there was trailing poop.

Bonus point that it can't be hit through consensus.